### PR TITLE
Update to julia1.6 & fix build issues

### PR DESCRIPTION
**Pull request by Islandman93**
Fixed issues with using the latest CxxWrap (v0.11.2) and switched to using Artifacts for ViZDoom source. I haven't fully tried to use this for experiments but it builds and the tests run (episodes + screen buffer + config, etc.). This also fixes issues with enums (or at least I tested they are correct) #7. Also bumped the version number to 0.1.1.

I can't figure out how to correctly get the labels object from GameState (line 300 ViZDoomJuliaModule). So I ended commenting that out.

### CODE REVIEW

**Comment by Islandman93** [comment]
Not sure why the test is failing, `RUN julia -e 'using Pkg; Pkg.add(path="./")'` works in the dockerfile see #11.

File: deps/lib_julia/ViZDoomJuliaModule.cpp [hunk 1]
```diff
@@ -312,16 +286,21 @@ JLCXX_MODULE ViZDoom(jlcxx::Module &mod)
...
+        .method("game_variables", [](GameState &gs) { return jlcxx::ArrayRef<double, 1>(gs.gameVariables.data(), gs.gameVariables.size()); })
         .method("screen_buffer", [](GameState &gs) { return gs.screenBuffer; })
         .method("depth_buffer", [](GameState &gs) { return gs.depthBuffer; })
         .method("labels_buffer", [](GameState &gs) { return gs.labelsBuffer; })
-        .method("automap_buffer", [](GameState &gs) { return gs.automapBuffer; })
-        .method("labels", [](GameState &gs) { return jlcxx::ArrayRef<Label, 1>(&(gs.labels[0]), gs.labels.size()); });
+        .method("automap_buffer", [](GameState &gs) { return gs.automapBuffer; });
+        // TODO: figure out how to return this? The below code breaks
+        // .method("labels", [](GameState &gs) { 
+        //     return jlcxx::ArrayRef<Label, 1>(labels.data(), labels.size());
```
Comment (findmyway) [inline deps/lib_julia/ViZDoomJuliaModule.cpp:302]: https://github.com/JuliaInterop/CxxWrap.jl#reference-native-julia-arrays

Maybe give it a try to `*Label` ?

File: deps/lib_julia/ViZDoomJuliaModule.cpp [hunk 2]
```diff
@@ -358,6 +338,7 @@ JLCXX_MODULE ViZDoom(jlcxx::Module &mod)
             return dg.makeAction(data, tics);
         })
         .method("advance_action", &DoomGame::advanceAction)
+        // TODO: What does this do that the above line doesn't?
```
Comment (findmyway) [inline deps/lib_julia/ViZDoomJuliaModule.cpp:341]: Seems like a duplication

File: test/basic.jl [hunk 3]
```diff
@@ -12,9 +13,16 @@ for i in 1:episodes
...
     while !ViZDoom.is_episode_finished(game)
         state = ViZDoom.get_screen_buffer(game)
-        print(size(state))
+        screen_shape = ViZDoom.get_screen_width(game), ViZDoom.get_screen_height(game), ViZDoom.get_screen_channels(game)
+        print(size(state), screen_shape)
+        # example reshaping screen size
+        image_state = reshape(state, screen_shape)
         r = ViZDoom.make_action(game, rand(actions))
         println("Reward $r")
-        sleep(sleep_time)
```
Comment (findmyway) [inline test/basic.jl:18]: Without sleep I remember the game will run very quickly……

**Review by findmyway — APPROVED** [review]
Nice work! I haven't got enough time to verify it but I'll merge it first.

**Comment by Islandman93** [comment]
Didn't realize I had made this from my master branch, fixing now.

File: test/basic.jl (hunk 3 above)
Comment (Islandman93) [reply test/basic.jl:18]: Yes this is just for the runtests.jl though so faster is better

**Pull request merged**